raw_result_diff.py

```diff
@@ -0,0 +1,198 @@
+# Copyright (c) 2021, Matthew Olsson <mattco@serenityos.org>
+#
+# SPDX-License-Identifier: MIT
+
+import json
+from argparse import ArgumentParser
+from pathlib import Path
+from typing import Tuple
+
+from main import TestResult
+from main import EMOJIS
+
+
+def read_result_files(old_path: Path, new_path: Path) -> Tuple[float, dict, dict]:
+    with old_path.open() as f:
+        old_results = json.loads(f.read())
+    with new_path.open() as f:
+        new_results = json.loads(f.read())
+
+    duration_delta = float(new_results["duration"]) - float(old_results["duration"])
+    old_results = old_results["results"]
+    new_results = new_results["results"]
+
+    return duration_delta, old_results, new_results
+
+
+class ResultParser:
+    def __init__(self, old_path: Path, new_path: Path, regressions: bool) -> None:
+        (duration_delta, old_results, new_results) = read_result_files(
+            old_path, new_path
+        )
+        self.duration_delta = duration_delta
+        self.old_results = old_results
+        self.new_results = new_results
+        self.regressions = regressions
+
+        self.new_tests: dict = {}
+        self.removed_tests: dict = {}
+        self.diff_tests: dict = {}
```

Comment:
Can we be more specific with the types here (and below)? `dict` on its own is already implied, the key and value types are what's interesting.